crates/sui-graphql-rpc/src/types/transaction_block.rs

```diff
@@ -438,27 +556,79 @@ impl TransactionBlockFilter {
             })?,
         })
     }
+
+    /// Returns count of filters that would require a `scan_limit` if used together.
+    pub(crate) fn num_filters(&self) -> usize {
```

Comment:
nit: maybe change the name of this function to better reflect that it's counting only certain filters?